packages/foundation/src/slots.tsx

```diff
@@ -42,6 +42,8 @@ export function withSlots<P>(
 ): React.ReactElement<P> | JSX.Element | null {
   const slotType = type as ISlot<P>;
   if (slotType.isSlot) {
+    // TODO: There is something weird going on here with children embedded in props vs. rest args.
```

Comment:
weird... curious what it could be, seems like you could always just execute the function...

Reply:
I need to resolve all of these TODOs together. I'll deal with that later, I just haven't yet since I have noticed any negative side effects